move remaining "theme" settings from settings.py dictionary to SiteMode model

tobiasmcnulty opened the pull request:
Lilia would like the ability to edit various strings on the site without re-deploying and without engaging a developer. This PR moves the remaining settings from `SITE_THEMES` into the `SiteMode` model.

tobiasmcnulty committed:
move remaining site-specific settings from the SITE_THEMES settings dictionary to the SiteMode model

tobiasmcnulty committed:
restore SITE_THEME_NAME to deployment local_settings.py

tobiasmcnulty commented:
@ejucovy Could you give this a review when you have a moment?

ejucovy commented on opendebates/models.py:
```diff
@@ -45,20 +46,51 @@ class SiteMode(CachingMixin, models.Model):
+
+    @property
+    def site_mode(self):
+        # avoid lots of unneeded round trips to memcached in production
```
Is there any way to stash this on the class instead of the instance?  Or even on the `get_site_mode` function object?  If I'm following correctly, even after this we'll still be making N queries (instead of ~5N) when we're rendering a template with N questions on it.

tobiasmcnulty replied:
Good catch! I would like to avoid caching anything on a Python object that may persist in the interpreter outside the request/response cycle (as doing so may lead to some difficult to troubleshoot invalidation issues), however, we already have the site mode in the template context so it's just a matter of figuring out how to get it into the `Submission` model methods (or how to refactor them/put them elsewhere so that's not required).

For the time being, I added a template tag which pre-fills `_cached_site_mode` with the `SITE_MODE` available in the template context, along with a test to make sure the link doesn't get broken.

Longer term, this probably needs some refactoring, but I think this will work for now.

ejucovy commented on opendebates/models.py:
```diff
@@ -125,13 +157,13 @@ def get_absolute_url(self):
-        text = settings.SITE_THEME['TWITTER_QUESTION_TEXT']
+        text = self.site_mode.twitter_question_text
         if self.voter.twitter_handle:
             text += u" h/t @%s" % self.voter.twitter_handle
```
While we're in here -- I know Lilia also wants to be able to adjust this `"h/t"` suffix behavior too, and put the submitter's twitter handle within the body of the tweet instead of at the end.  ("Vote on @SO-AND-SO's question!")  But that gets complicated by the fact that twitter handle is optional, so there has to be some branching logic somewhere.  What would you think about having two fields on site mode here instead -- `twitter_question_text_with_handle` and `twitter_question_text_no_handle`?   Then this method would be something like

```
  if self.voter.twitter_handle:
    return self.site_mode.twitter_question_text_with_handle.format(handle=self.voter.twitter_handle)
  return self.site_mode.twitter_question_text_no_handle
```

tobiasmcnulty committed:
add missing ] in default facebook question title

tobiasmcnulty committed:
pre-fill _cached_site_mode on Submission objects in list_ideas.html to avoid a call out to the cache for each and every idea

tobiasmcnulty committed:
add test showing get_side_mode called only once by list_ideas

tobiasmcnulty committed:
allow specifying different twitter_question_text depending on whether or not the voter supplied a twitter handle

tobiasmcnulty committed:
modify test for site_mode caching in list_ideas so it works when called with other tests

tobiasmcnulty committed:
allow placing h/t twitter handle at any location

tobiasmcnulty committed:
Merge branch 'develop' into move_site_theme_settings_to_site_mode

tobiasmcnulty commented:
@ejucovy Do you want to take another look at this? I think I've addressed all your feedback. I may go ahead and merge in an hour or two if I haven't heard back so Lilia can test / get production content ready. We can take up any issues you find as new work if that's okay (please do make sure I implemented your suggested changes appropriately!).

ejucovy commented:
All looks good to me!

tobiasmcnulty committed:
break up all the social sharing content sections a little more sensibly in the admin

tobiasmcnulty committed:
remove unused Submission.my_tweet_text method
